routes/api/articles.js

```diff
@@ -0,0 +1,15 @@
+import express from 'express';
+import articlesController from '../../controllers/articles';
+import validation from '../../middlewares/articleValidation';
+
+const router = express.Router();
+
+router.get('/article/feeds', articlesController.getFeeds);
+router.get('/articles', articlesController.getAllPublished);
+router.get('/articles/draft', articlesController.getAllDraft);
+router.post('/articles', validation.article, articlesController.createAnArticle);
+router.delete('/articles/:slug', validation.slug, articlesController.deleteArticle);
```

Comment:
I think it is not good to let any User is able to delete the article. some sort of authentication is needed to access these routes.

Reply:
@shemaeric The is a bug coming from the way you used the jwt passport strategy, it handles only local logins but in a user logins in with social, it throws an error because the user id is different from the one we have in the database. I think it's better we use the email or username because these are unique too.

Reply:
Actually, the way jwt strategy works is that it extracts the token from headers and use the id from the payloads of the token and search into the database to check if the user with that id exists, if the user exists he will be authorized and if not, you will get the response object with the error message says that, that user does not exists. When using social login, It we keep the new id of the user and the email and creates the token based on id and also the email, so the jwt strategy has nothing to do with the errors you are getting.

Reply:
This has been resolved.